src/detect-engine-analyzer.c

```diff
@@ -701,10 +701,10 @@ static void DumpContent(JsonBuilder *js, const DetectContentData *cd)
         jb_set_uint(js, "depth", cd->depth);
     }
     if (cd->flags & DETECT_CONTENT_DISTANCE) {
-        jb_set_uint(js, "distance", cd->distance);
+        jb_set_int(js, "distance", cd->distance);
```

Comment:
How come there were no warnings ?

Reply:
`Wimplicit-int-conversion` is only set in one build and this `DumpContent` is not always active ?

Reply:
No idea, the code compilation is not conditional.

Reply:
https://redmine.openinfosecfoundation.org/issues/6632

Reply:
`Wsign-conversion` is the warning we want